vignettes/data-as-shiny-module.Rmd

```diff
@@ -0,0 +1,127 @@
+---
+title: "Data as shiny Module"
+author: "NEST CoreDev"
+output: 
+  rmarkdown::html_vignette:
+    toc: true
+vignette: >
+  %\VignetteIndexEntry{Data as shiny Module}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+## Introduction
+
+Proper functioning of any `teal` application requires presence of a `teal_data` object.
+Typically, a `teal_data` object created in the global environment will be passed to the `data` argument in `init`.
+This `teal_data` object should contain all elements necessary for successful execution of the application's modules. 
+In some scenarios, however, application developers may opt to postpone some data operations until the application run time.
+This can be done by passing a special _`shiny` module_ to the `data` argument.
+The `teal_data_module` function is used to build such a module from the following components:
+
+- a `ui` function; accepts only one argument, `id`; defines user interface elements for the data module
+- a `server` function: accepts only one argument, `id`; defines server logic for the data module, including data creation; must return a reactive expression containing a `teal_data` object
+
+`teal` will run this module when the application starts and the resulting `teal_data` object that will be used throughout all `teal` (analytic) modules. 
+
+## Creating Data In-App
+
+One case for postponing data operations are data sets that are dynamic, frequently updated.
+Such data cannot be created once and kept in the global environment.
+Using `teal_data_module` allows to create a data set from scratch every time the user starts the application.
+This the user will always have access to the most recent version of the data.
```

Comment:
```suggestion
This way, the user will always have access to the most recent version of the data.
```

Reply:
Also: I don't understand this. What is "version of the data"? If the data is static (no change in time) - how the modules assures that it's "most recent".

I feel that what you mean is: "data will be recreated for each application session". But you already have such statement before. So maybe add: "In some of the cases, this will assure that the most recent version of the data is used".

Reply:
If you read the whole paragraph, not every sentence in isolation, the last sentence clearly refers to a dynamic data situation. Or is this my author bias?

Reply:
I've removed last sentence and seems to be fine.

Reply:
I'm ok with this. My vignettes comments are not that _strong_ anyway.